src/main/java/com/apuope/apuope_re/services/LogInService.java

```diff
@@ -5,29 +5,36 @@
 import com.apuope.apuope_re.repositories.UserRepository;
 import com.apuope.apuope_re.dto.UserCredentials;
 import org.jooq.DSLContext;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
 import org.springframework.stereotype.Service;
 import java.util.Optional;
 
 @Service
 public class LogInService {
     private final DSLContext dslContext;
     private final UserRepository userRepository;
+    private final BCryptPasswordEncoder passwordEncoder;
+
+    @Autowired
+    private JWTService jwtService;
 
     public LogInService(DSLContext dslContext, UserRepository userRepository) {
         this.dslContext = dslContext;
         this.userRepository = userRepository;
+        this.passwordEncoder = new BCryptPasswordEncoder(10);
     }
 
     public ResponseData<String> validateUser(UserCredentials userCredentials) {
         Optional<UsersRecord> userOpt =
-                userRepository.findVerifiedUserByEmail(userCredentials.getEmail(), this.dslContext);
```

Comment:
This should work when there is allowed calls to be made in /api/verify in SecurityConfig. We should allow only verified users to log in.